File: app/js_src/components/context.js
// vim: set ft=javascript.jsx:
import React from 'react'
import _ from 'lodash'

class Context extends React.Component {
  render() {
    return (
      <div className="context">
        <table>
          <tbody>
            { _.map(this.props.params, (param) => <tr><td>{param.name}</td><td><input type="text" value={param.value} /></td></tr>) }
          </tbody>
        </table>
      </div>
    );
  }
}

export default Context

File: app/models/generator/context.rb
module Generator
  class Context
    def self.from_json(doc)
      kvs = doc[:params].map do |param|
        case param[:type]
        when 'text' then [param[:name], param[:value]]
        when 'image' then [param[:name], param[:value]]
        end
      end
      Hash[kvs]
    end
  end
end
